File: SupplierPortal.Web/Modules/Inventory/Item/ItemColumns.cs
using Serenity.ComponentModel;
using SupplierPortal.Inventory.Enum;
using System.ComponentModel;

namespace SupplierPortal.Inventory.Columns;

[ColumnsScript("Inventory.Item")]
[BasedOnRow(typeof(ItemRow), CheckNames = true)]
public class ItemColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    [Visible (false)]
    public int Id { get; set; }
    [EditLink]
    public string ItemCode { get; set; }
    [EditLink]
    public string ItemName { get; set; }
    public KDV KDV { get; set; }
    public Unit Unit { get; set; }
}

File: SupplierPortal.Web/Modules/ServerTypes/Inventory/ItemColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { KDV } from "./Enum.KDV";
import { ItemRow } from "./ItemRow";

export interface ItemColumns {
    Id: Column<ItemRow>;
    ItemCode: Column<ItemRow>;
    ItemName: Column<ItemRow>;
    KDV: Column<ItemRow>;
    Unit: Column<ItemRow>;
}

export class ItemColumns extends ColumnsBase<ItemRow> {
    static readonly columnsKey = 'Inventory.Item';
    static readonly Fields = fieldsProxy<ItemColumns>();
}

[KDV]; // referenced types